### PR TITLE
chapter 7

### PR DESCRIPTION
**Pull request by starocketeer**

### CODE REVIEW

**Comment by starocketeer** [comment]
Ready

File: app/assets/stylesheets/users.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Users controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: https://sass-lang.com/
```
Comment (saobv-2223) [inline app/assets/stylesheets/users.scss]: remove file that never used

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -1,3 +1,27 @@
...
+        end
+      end
+    
+      private
+    
+      def user_params
+        params.require(:user).permit :name, :email, :password, :password_confirmation
+      end
+    end
+
```
Comment (saobv-2223) [inline app/controllers/users_controller.rb]: indenting only 2 spaces, please

File: app/helpers/users_helper.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+module UsersHelper
+def gravatar_for user
```
Comment (saobv-2223) [inline app/helpers/users_helper.rb]:
```suggestion
  def gravatar_for user
```

File: config/routes.rb [hunk 4]
```diff
@@ -1,10 +1,19 @@
 Rails.application.routes.draw do
   root "static_pages#home"
   
+<<<<<<< HEAD
```
Comment (saobv-2223) [inline config/routes.rb]: resolve all the conflict, please

**Comment by starocketeer** [comment]
Ready

File: app/assets/stylesheets/users.scss (hunk 1 above)
Comment (starocketeer) [reply app/assets/stylesheets/users.scss]: Done

File: app/controllers/users_controller.rb (hunk 2 above)
Comment (starocketeer) [reply app/controllers/users_controller.rb]: Done

File: app/helpers/users_helper.rb (hunk 3 above)
Comment (starocketeer) [reply app/helpers/users_helper.rb]: Done

File: config/routes.rb (hunk 4 above)
Comment (starocketeer) [reply config/routes.rb]: Done

**Comment by starocketeer** [comment]
Ready

**Comment by ngocvt-0484** [comment]
merged

**Pull request merged**